Shared/Extensions/UIImageExtensions.swift

```diff
@@ -21,44 +21,44 @@ extension UIImage {
 
     /// Generates a UIImage from GIF data by calling out to SDWebImage. The latter in turn uses UIImage(data: NSData)
     /// in certain cases so we have to synchronize calls (see bug 1223132).
-    public static func imageFromGIFDataThreadSafe(data: NSData) -> UIImage? {
+    public static func imageFromGIFDataThreadSafe(_ data: Data) -> UIImage? {
         imageLock.lock()
-        let image = UIImage.sd_animatedGIFWithData(data)
+        let image = UIImage.sd_animatedGIF(with: data)
         imageLock.unlock()
         return image
     }
 
-    public static func dataIsGIF(data: NSData) -> Bool {
-        guard data.length > 3 else {
+    public static func dataIsGIF(_ data: Data) -> Bool {
+        guard data.count > 3 else {
             return false
         }
 
         // Look for "GIF" header to identify GIF images
-        var header = [UInt8](count: 3, repeatedValue: 0)
-        data.getBytes(&header, length: 3 * sizeof(UInt8))
+        var header = [UInt8](repeating: 0, count: 3)
+        (data as NSData).getBytes(&header, length: 3 * MemoryLayout<UInt8>.size)
```

Comment:
apparently, according to StackOverflow anyway, the way to do this in Swift 3.0 is

`data.copyBytes(to: &header, count: 3 * MemoryLayout<UInt8>.size)`